source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -175,7 +175,10 @@ void AuthenticatorImpl::startVerify() {
   if (jwks_data_->getJwtProvider().clock_skew_seconds() > 0) {
     clock_skew_seconds = jwks_data_->getJwtProvider().clock_skew_seconds();
   }
-  status = jwt_->verifyTimeConstraint(absl::ToUnixSeconds(absl::Now()), clock_skew_seconds);
+  const uint64_t unix_timestamp =
+      std::chrono::duration_cast<std::chrono::seconds>(timeSource().systemTime().time_since_epoch())
```

Comment:
WDYT of factoring this out into `DateUtil::nowToSeconds()`. There is already `DateUtil::nowToMilliseconds()` which is very close to what this code looks like.

See source/common/common/utility.cc around line 260.

Reply:
@jmarantz 
What are your thoughts about adding methods to the TimeSource interface that return absl::Time so methods like absl::ToUnixSeconds can be applied to them?

Reply:
I'm fine with that as long as we are ok putting it on the base class. The helper function pattern here predated my involvement.